dev/tools/bin/generate_gradle_lockfiles.dart

```diff
@@ -2,29 +2,40 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// This script generates `android/build.gradle` for each directory specify in the stdin.
+// For each directory specified in the stdin, this script generates:
+//     1. The top-level build.gradle (android/build.gradle).
+//     2. The top level settings.gradle (android/settings.gradle).
+//     3. The gradle wrapper file (android/gradle/wrapper/gradle-wrapper.properties).
 // Then it generate the lockfiles for each Gradle project.
-// To regenerate these files, run `find . -type d -name 'android' | dart dev/tools/bin/generate_gradle_lockfiles.dart`
+// To regenerate these files, run `find . -type d -name 'android' | dart dev/tools/bin/generate_gradle_lockfiles.dart`.
 
+import 'dart:collection';
 import 'dart:io';
 
 import 'package:args/args.dart';
 import 'package:file/file.dart';
 import 'package:file/local.dart';
 import 'package:path/path.dart' as path;
+import 'package:yaml/yaml.dart';
 
 void main(List<String> arguments) {
   const String usageMessage = "Usage: find . -type d -name 'android' | dart dev/tools/bin/generate_gradle_lockfiles.dart\n"
       'If you would rather enter the files manually, just run `dart dev/tools/bin/generate_gradle_lockfiles.dart`,\n'
       "enter the absolute paths to the app's android directory, then press CTRL-D.\n"
       "If you don't wish to re-generate the settings.gradle, build.gradle, and gradle-wrapper.properties files,\n"
-      "add the flag '--no-gradle-generation'";
+      'add the flag `--no-gradle-generation`.\n'
+      'Automatically excludes a set of android subdirectories, defined at dev/tools/bin/config/lockfile_exclusion.yaml.\n'
```

Comment:
Should this instead say "This tool automatically ..."

Reply:
changed!